src/components/SettingsDialog.vue

```diff
@@ -1,19 +1,24 @@
 <template>
-    <AppNavigationSettings :open="true">
-        <div id="app-settings">
+    <AppSettingsDialog
+        :open.sync="isOpen"
+        :title="t('cookbook', 'Cookbook settings')"
+        :show-navigation="true"
+        first-selected-section="keyboard shortcuts"
+    >
+        <AppSettingsSection
+            :title="t('cookbook', 'Recipe folder')"
+            class="app-settings-section"
+        >
             <fieldset>
                 <ul>
                     <li>
-                        <ActionButton
-                            class="button"
-                            :icon="
-                                scanningLibrary
-                                    ? 'icon-loading-small'
-                                    : 'icon-history'
-                            "
-                            :title="t('cookbook', 'Rescan library')"
-                            @click="$emit('reindex')"
-                        />
+                        <NcButton @click="$emit('reindex')">
```

Comment:
The button is causing a JS exception on my machine when clicking it. Can you confirm?

Reply:
I think this should be fixed in c9d37206. I had to replace `$emit('reindex')` with `reindex` because now that method is in the same component.

Reply:
Does no longer happen on my machine. So it seems to be fixed in the meantime.